charts/opentelemetry-collector/values.schema.json

```diff
@@ -145,6 +145,12 @@
     "priorityClassName": {
       "type": "string"
     },
+    "initContainers": {
+      "type": "array",
+      "items": {
+        "type": "object"
```

Comment:
Is there ever a time an object in this array would contain more properties than:

```yaml
  - name: test
    command:
      name: cp
      extraArgs:
      - /bin/sleep
      - /test/sleep
    image:
      repository: busybox
      pullPolicy: IfNotPresent
      tag: latest
```

If not, I'd opt to enforce this object's properties.

Reply:
I added the properties to the schema. It's entirely possible that someone might require env variables or perhaps mounting secrets to the init container, but in that case the pod construct would need to be changed to add support for them, so I would say this is enough for now.

Reply:
The current version of this PR matches my use case, can't speak for anyone else. Secrets and env vars are just the things that I've ran into when dealing with init containers previously, but obviously even those won't handle 100% of the possible use cases there are.

I can add support for container schema. What levels should I add: just the top level properties, or all they way down?

Reply:
@TylerHelmuth, why `initContainers.enabled` is needed? can't we can check if `initContainers` list is empty instead?

Reply:
We could, and if this is the only place it would get used (not part of a preset or used in some other decision making somewhere) then that would be best.  It is in line with some other objects in the values.yaml.

Reply:
Thanks for the comments, I will update this pull request tomorrow with the schema changes.

Reply:
Understood. I'll make the relevant changes there as well.

Reply:
I updated the `initContainers` schema to be a list of objects, which only require `name` property (which seems to be the only required property for a container) but allow any other ones. The objects are rendered in `_pod.tlp` using `toYaml` if the `initContainers` list is not empty.